core/src/main/java/org/wildfly/plugin/core/GalleonUtils.java

```diff
@@ -111,17 +111,43 @@ public static ProvisioningConfig buildConfig(Path provisioningFile) throws Provi
      * @param layers Layers to include.
      * @param excludedLayers Layers to exclude.
      * @param pluginOptions Galleon plugin options.
+     * @param log Log to log messages.
      * @return The provisioning config.
      * @throws ProvisioningException
      */
     public static ProvisioningConfig buildConfig(ProvisioningManager pm,
             List<FeaturePack> featurePacks,
             List<String> layers,
             List<String> excludedLayers,
-            Map<String, String> pluginOptions) throws ProvisioningException, IllegalArgumentException {
+            Map<String, String> pluginOptions, LogConsumer log) throws ProvisioningException, IllegalArgumentException {
         final ProvisioningConfig.Builder state = ProvisioningConfig.builder();
-        for (FeaturePack fp : featurePacks) {
+        boolean hasLayers = !layers.isEmpty();
+        boolean fpWithDefaults = true;
+        if (!hasLayers) {
+            // Check we have all feature-packs with default values only.
+            for (FeaturePack fp : featurePacks) {
+                if (fp.isInheritConfigs() != null ||
+                        fp.isInheritPackages() != null ||
+                        !fp.getIncludedConfigs().isEmpty() ||
+                        !fp.getExcludedConfigs().isEmpty() ||
+                        fp.isTransitive() ||
+                        !fp.getExcludedPackages().isEmpty() ||
+                        !fp.getIncludedPackages().isEmpty()) {
+                    fpWithDefaults = false;
+                    break;
+                }
+            }
+            if (fpWithDefaults) {
+                log.log(LogConsumer.Type.INFO, "No Layers have been provided and feature-packs don't override default "
+                        + "packages and configs inheritance. An attempt is done to provision the last feature-pack default configuration. "
+                        + "In case that is not what you would expect, make sure to properly inherit configurations "
+                        + "from the right feature-pack using <inherit-configs>true|false</inherit-configs>");
+            }
+        }
 
+        for (int i = 0; i < featurePacks.size(); i++) {
+            FeaturePack fp = featurePacks.get(i);
+            boolean isLastFP = (i == featurePacks.size() - 1);
```

Comment:
So last is really last as mentioned in list of FPs. So when I put to list also datasource feature pack, this default mechanism will be broken?

Is it dumped somewhere in plugin run what `inherit-*` were in fact used?

Reply:
Yes, don't put the datasources as the last one. But the datasources shouldn't be used, it has just layers, no default configs.

Reply:
Just for me to understand. So `inherit-config=true` cant be set on datasources? Or can but will be ignored or do nothing.

So logic here when layers.isEmpty() then mark `inherit-config=true` on last FP will ignore datasource fp?

Reply:
Yes, it could be set, but nothing will get provisioned, there are no default configs defined in this feature-pack. 
We can't ignore an FP, we have no knowledge of the FP. If the datasources FP is the latest, nothing will get proviioned with the default logic.

Reply:
Ok got it. So therefore this would be important to troubleshoot similar cases

`Is it dumped somewhere in plugin run output what values of inherit-* were in fact used?`

Reply:
Yes, the information can be found in the server. if record-provisioning-state is set to true, then galleon generates a .galleon/provisioning.xml file.
If record-provisioning-state is not set (the default), we generate a .wildfly-maven-plugin-provisioning.xml file that contains this information.